raster/r.random.surface/testsuite/test_r_random_surface.py

```diff
@@ -0,0 +1,284 @@
+        self.assertTrue(0 <= mean <= 255, "Mean outside expected range")
+        self.assertGreater(stddev, 0, "No variation in output values")
+
+    def test_custom_high(self):
+        """Test with custom high value."""
+        high_val = 100
+        self.runModule(
+            "r.random.surface", output=self.output_raster, high=high_val, overwrite=True
+        )
+
+        stats_json = gs.read_command(
+            "r.univar", map=self.output_raster, flags="g", format="json"
+        )
+        stats = json.loads(stats_json)
+
+        actual_min = float(stats[0]["min"])
+        actual_max = float(stats[0]["max"])
+
+        self.assertGreaterEqual(
+            actual_min, 0, f"Minimum value {actual_min} is less than 0"
+        )
+        self.assertLessEqual(
+            actual_max,
+            high_val,
+            f"Maximum value {actual_max} is greater than specified high value {high_val}",
+        )
+
+    def test_regional_correlation(self):
+        """Test spatial correlation properties using r.stats"""
+        self.runModule(
+            "r.random.surface",
+            output=self.output_raster,
+            distance=5.0,
+            exponent=3.0,
+            seed=42,
+            overwrite=True,
+        )
+
+        output_stats = f"{self.output_raster}_stats"
+        self.runModule(
+            "r.mapcalc",
+            expression=f"{output_stats} = {self.output_raster}[-1,-1]",
+            quiet=True,
+        )
+
+        stats_output = gs.read_command(
+            "r.stats",
+            input=f"{self.output_raster},{output_stats}",
+            flags="c",
+            quiet=True,
+        ).splitlines()
+
+        self.assertGreater(
+            len(stats_output), 0, "r.stats -c output is empty or malformed"
+        )
+
+        frequencies = [line.split() for line in stats_output if len(line.split()) == 3]
+
+        self.assertGreater(
+            len(frequencies), 0, "No valid frequency data extracted from r.stats -c"
+        )
+
+        values = np.array([float(f[0]) for f in frequencies])
+        counts = np.array([int(f[2]) for f in frequencies])
+
+        correlation = np.corrcoef(values, counts)[0, 1]
+        self.assertGreater(
+            correlation,
+            0,
+            "No positive spatial correlation detected with distance parameter",
+        )
+
+        self.runModule(
+            "g.remove", type="raster", name=output_stats, flags="f", quiet=True
+        )
+
+    def test_uniform_distribution(self):
+        """Test that -u flag produces uniformly distributed values."""
+        self.runModule(
+            "r.random.surface",
+            output=self.output_raster,
+            seed=42,
+            flags="u",
+            overwrite=True,
+        )
+
+        stats_json = gs.read_command(
+            "r.univar", map=self.output_raster, flags="g", format="json"
+        )
+        stats = json.loads(stats_json)
```

Comment:
This can be shortened using the parse_command:
```suggestion
        stats = gs.parse_command("r.univar", map=self.output_raster, format="json")
```

Reply:
Updated the file with the suggested changes and removed float conversions in the following lines.

Reply:
Great, but there are 3 other cases like that.

Reply:
I am sorry, I had overlooked the other cases. I have now updated the tests to use the parse command.